pynucastro/networks/rate_collection.py

```diff
@@ -636,44 +637,73 @@ def _build_collection(self):
         self.custom_rates = []
         self.approx_rates = []
         self.derived_rates = []
+        self.modified_rates = []
 
         for r in self.rates:
             if isinstance(r, ApproximateRate):
                 self.approx_rates.append(r)
                 for cr in r.get_child_rates():
                     assert cr.chapter != "t"
-                    # child rates may be ReacLibRates or DerivedRates
-                    # make sure we don't double count
-                    if isinstance(cr, DerivedRate):
 
-                        # Here we check whether this child rate is removed or not.
-                        # removed means that this rate is never used on its own to connect two nuclei in the network
-                        # it is only used in one or more ApproximateRate.
-                        if cr not in self.rates:
-                            cr.removed = True
-                        else:
-                            cr.removed = False
+                    # Check whether this child rate is removed or not.
+                    # "removed" means that this rate is never used on
+                    # its own to connect two nuclei in the network it
+                    # is only used in one or more ApproximateRate or
+                    # ModifiedRate
+                    if cr not in self.rates:
+                        cr.removed = True
+                    else:
+                        cr.removed = False
 
-                        cr.fname = None
-                        # pylint: disable-next=protected-access
-                        cr._set_print_representation()
+                    cr.fname = None
+                    # pylint: disable-next=protected-access
+                    cr._set_print_representation()
 
+                    # child rates may be ReacLibRates, ModifiedRates,
+                    # or DerivedRates.  Make sure we don't double
+                    # count
+                    if isinstance(cr, DerivedRate):
                         if cr not in self.derived_rates:
                             self.derived_rates.append(cr)
-
+                    elif isinstance(cr, ModifiedRate):
+                        if cr not in self.derived_rates:
+                            self.modified_rates.append(cr)
                     else:
-                        if cr not in self.rates:
-                            cr.removed = True
-                        else:
-                            cr.removed = False
-
-                        cr.fname = None
-                        # pylint: disable-next=protected-access
-                        cr._set_print_representation()
-
                         if cr not in self.reaclib_rates:
                             self.reaclib_rates.append(cr)
 
+            elif isinstance(r, ModifiedRate):
+                self.modified_rates.append(r)
```

Comment:
do we need to check if `r` is already in `modified_rates` here since we assumed child rates for `approx_rates` can potentially be a `ModifiedRate`.

Reply:
I guess technically we do.  I can't imagine this being common though.  I'll add a check.